.github/workflows/build-docker.yml

```diff
@@ -0,0 +1,41 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches: [main, 'dpe*', 'DPE*']
```

Comment:
Do you want to add some concurrency handling to prevent issues from the same branch running multiple jobs at once, or do you think this isn't a risk atp?

Here is how this is handled for the test clone workflow on Snowflake:

https://github.com/Sage-Bionetworks/snowflake/blob/ad8edd09cba2e86cd66156924e39557accfdbfd5/.github/workflows/test_with_clone.yaml#L13